packages/block-library/src/tabs/view.js

```diff
@@ -0,0 +1,218 @@
+					if ( tabIndex !== null ) {
+						privateActions.setActiveTab( tabIndex + 1 );
+					}
+				} else if ( event.key === 'ArrowLeft' && ! isVertical ) {
+					const { tabIndex } = privateState;
+					if ( tabIndex !== null ) {
+						privateActions.setActiveTab( tabIndex - 1 );
+					}
+				} else if ( event.key === 'ArrowDown' && isVertical ) {
+					const { tabIndex } = privateState;
+					if ( tabIndex !== null ) {
+						privateActions.setActiveTab( tabIndex + 1 );
+					}
+				} else if ( event.key === 'ArrowUp' && isVertical ) {
+					const { tabIndex } = privateState;
+					if ( tabIndex !== null ) {
+						privateActions.setActiveTab( tabIndex - 1 );
+					}
+				}
+			} ),
+			/**
+			 * Handles the click event for the tab label.
+			 *
+			 * @param {MouseEvent} event The click event.
+			 */
+			handleTabClick: withSyncEvent( ( event ) => {
+				event.preventDefault();
+
+				const { tabIndex } = privateState;
+				if ( tabIndex !== null ) {
+					privateActions.setActiveTab( tabIndex );
+				}
+			} ),
+			/**
+			 * Sets the active tab index (internal implementation).
+			 *
+			 * @param {number}  tabIndex    The index of the active tab.
+			 * @param {boolean} scrollToTab Whether to scroll to the tab element.
+			 */
+			setActiveTab: ( tabIndex, scrollToTab = false ) => {
+				const context = getContext();
+				context.activeTabIndex = tabIndex;
+				if ( scrollToTab ) {
+					const tabId = privateState.tabsList[ tabIndex ].id;
+					const tabElement = document.getElementById( tabId );
+					if ( tabElement ) {
+						setTimeout( () => {
+							tabElement.scrollIntoView( { behavior: 'smooth' } );
+						}, 100 );
+					}
+				}
+			},
+		},
+		callbacks: {
+			/**
+			 * When the tabs are initialized, we need to check if there is a hash in the url and if so if it exists in the current tabsList, set the active tab to that index.
+			 *
+			 */
+			onTabsInit: () => {
+				const { tabsList } = privateState;
+				if ( tabsList.length === 0 ) {
+					return;
+				}
+				const { hash } = window.location;
+				const tabId = hash.replace( '#', '' );
+				const tabIndex = tabsList.findIndex( ( t ) => t.id === tabId );
+				// Check if tabIndex is a positive number and if so we'll auto activate that tab.
+				if ( tabIndex >= 0 ) {
+					privateActions.setActiveTab( tabIndex, true );
+				}
+			},
+		},
+	},
+	{
+		lock: true,
+	}
+);
+
+// Public store for third-party extensibility.
+store( 'core/tabs', {
+	state: {
+		/**
+		 * Gets a contextually aware list of tabs for the current tabs block.
+		 * Public API for third-party access.
+		 *
+		 * @type {Array}
+		 */
+		get tabsList() {
+			return createReadOnlyProxy( privateState.tabsList );
+		},
+		/**
+		 * Gets the index of the active tab element whether it
+		 * is a tab label or tab panel.
+		 * Public API for third-party access.
+		 *
+		 * @type {number|null}
+		 */
+		get tabIndex() {
+			return createReadOnlyProxy( privateState.tabIndex );
```

Comment:
You don't need a proxy if it's not an array or an object.


```suggestion
			return privateState.tabIndex;
```